[misc] Provide instructions about what to do when the libraries needed by the generate_compose_yaml.py script are missing

sdumitriu opened the pull request.

sdumitriu committed:
[misc] Provide instructions about what to do when the libraries needed by the generate_compose_yaml.py script are missing

sashaandjic commented:
![image](https://github.com/data-team-uhn/cards/assets/23243244/b64a33d9-318e-4619-a3ad-54a7d8cee756)

sashaandjic commented:
> Is this expected? ![image](https://user-images.githubusercontent.com/23243244/244111660-b64a33d9-318e-4619-a3ad-54a7d8cee756.png)

Seems to be fixed in https://github.com/data-team-uhn/cards/pull/1524

sdumitriu commented:
It is kind of expected, it just means you have an older version of the tzlocal library.

sashaandjic reviewed (approved):
tested

sdumitriu commented:
@sashaandjic this doesn't change anything when the libraries are already installed. To test it properly, you should first remove them with `pip uninstall tzlocal pyOpenSSL PyYAML`.

sashaandjic reviewed (commented):
tested 

missing tzlocal 
-----
![image](https://github.com/data-team-uhn/cards/assets/23243244/e117057b-6790-4328-b25a-d308902c8221)

missing pyOpenSSL
------
![image](https://github.com/data-team-uhn/cards/assets/23243244/63eb799f-0eff-4bca-8b7a-1cb48334e551)

missing PyYAML
-----
![image](https://github.com/data-team-uhn/cards/assets/23243244/474c59c6-eb5b-48a6-aa6c-24e72c72c9c3)

IntegralProgrammer committed:
[misc] Provide instructions about what to do when the libraries needed by the generate_compose_yaml.py script are missing

If at least one module fails to import, display instructions describing how to install all required modules

IntegralProgrammer committed:
[misc] Provide instructions about what to do when the libraries needed by the generate_compose_yaml.py script are missing

Fix requirements.txt

IntegralProgrammer reviewed (commented):
I've updated this branch so that if at least one of the Python dependency modules is missing, the user is shown instructions on how to install these packages, both by using `pip` and by `apt`.

sdumitriu commented on compose-cluster/requirements.txt:
```diff
@@ -0,0 +1,25 @@
+# specific language governing permissions and limitations
+# under the License.
+
+psutil==5.8.0
+pyOpenSSL==23.2.0
+cryptography==41.0.1
+cffi==1.15.1
+pycparser==2.21
+PyYAML==5.3.1
+tzlocal==2.1
+pytz==2023.3
```
Why such old versions? Do the newer ones not work on the version of python used in our containers?

IntegralProgrammer replied:
I picked versions that roughly matched what I had installed on my Debian 11 system via `apt`, but given that only 3 modules needed to be upgraded, I have made that change in https://github.com/data-team-uhn/cards/pull/1526/commits/b317cb4061cd10ac3db994581da2e43adf72d85a. have tested it, and can confirm that it works.

IntegralProgrammer committed:
[misc] Provide instructions about what to do when the libraries needed by the generate_compose_yaml.py script are missing

Upgrade Python packages listed in requirements.txt
